source/postprocess/mobility_statistics.cc

```diff
@@ -0,0 +1,201 @@
+#include <aspect/postprocess/mobility_statistics.h>
+#include <aspect/material_model/simple.h>
+#include <aspect/global.h>
+#include <aspect/utilities.h>
+#include <aspect/geometry_model/interface.h>
+#include <aspect/geometry_model/spherical_shell.h>
+
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/fe/fe_values.h>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    template <int dim>
+    std::pair<std::string,std::string>
+    MobilityStatistics<dim>::execute (TableHandler &statistics)
+    {
+      const QGauss<dim> quadrature_formula (this->get_fe()
+                                            .base_element(this->introspection().base_elements.velocities).degree+1);
+      const unsigned int n_q_points = quadrature_formula.size();
+
+      FEValues<dim> fe_values (this->get_mapping(),
+                               this->get_fe(),
+                               quadrature_formula,
+                               update_values   |
+                               update_quadrature_points |
+                               update_JxW_values);
+      std::vector<Tensor<1,dim>> velocity_values(n_q_points);
+
+      double local_velocity_square_integral = 0;
+
+      for (const auto &cell : this->get_dof_handler().active_cell_iterators())
+        if (cell->is_locally_owned())
+          {
+            fe_values.reinit (cell);
+            fe_values[this->introspection().extractors.velocities].get_function_values (this->get_solution(),
+                                                                                        velocity_values);
+            for (unsigned int q = 0; q < n_q_points; ++q)
+              {
+                local_velocity_square_integral += ((velocity_values[q] * velocity_values[q]) *
+                                                   fe_values.JxW(q));
+              }
+          }
+
+      const double global_velocity_square_integral
+        = Utilities::MPI::sum (local_velocity_square_integral, this->get_mpi_communicator());
+
+      const double vrms = std::sqrt(global_velocity_square_integral) /
+                          std::sqrt(this->get_volume());
+
+      //now compute surface vrms
+      // create a quadrature formula for the velocity.
+      const QGauss<dim-1> quadrature_formula_for_surface (this->introspection().polynomial_degree.velocities+1);
+  
+      FEFaceValues<dim> fe_face_values (this->get_mapping(),
+                                        this->get_fe(),
+                                        quadrature_formula_for_surface,
+                                        update_values |
+                                        update_JxW_values |
+                                        update_quadrature_points);
+      
+      std::vector<Tensor<1,dim>> velocities (fe_face_values.n_quadrature_points);
+      
+      std::map<types::boundary_id, double> local_velocity_square_integral_surface;
+      std::map<types::boundary_id, double> local_boundary_area;
```

Comment:
Maps are only necessary for the place where you copied this from, because it computes these values for all boundaries. Here you want two single values (rms velocity and area of the top surface). Therefore this will make a lot of things simpler:

```suggestion
      double local_velocity_square_integral_top_boundary;
      double local_top_boundary_area;
```